xformers/triton/k_activations.py

```diff
@@ -135,3 +137,29 @@ def gelu_grad(x):
     return 0.5 * x * (
         (1 - tanh_out * tanh_out) * (0.79788456 + 0.1070322243 * x * x)
     ) + 0.5 * (1 + tanh_out)
+
+
+@triton.jit
+def smelu(x, beta=2.0):
```

Comment:
I don't think that you can pass  a default param with triton actually, it only works with a subset of the python syntax and my guess is that this is out of it (cc @ptillet). Something could be worth trying, having a getter for this kernel, like the following

`
def get_smelu_kernel(beta: float = 2.0):
   @triton.jit
   def smelu(x):
      pass  # use beta here, but maybe that this will fail at the JIT phase
`

If that does not work, 
- for a start we could have a fixed beta, then iterate on the implementation to expose it (completely fine by me)
- could be that the activation kernel take another parameter, which in that case would be the beta value, or that we figure out with Phil how to generate the kernel code on the fly with the proper beta

Reply:
thanks @blefaudeux I'll give it a try... a bit late here so wanted to give it a shot in the morning 😴

Reply:
checking this with Philippe, the default value should work actually, maybe that it needs to be : float ? or similar

Reply:
ah ok! cool let me check

Reply:
To be clear, what should work (for now) is default arguments for tl.constexpr annotated arguments :p

Reply:
ah right... i'm on triton 1.x at the moment...

Reply:
we need to update to triton2.. CI is blocking right now, I hope to get that sorted out this week end